src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -572,27 +546,29 @@ function initRTDObserver() {
   observer.observe(document.body, config);
 }
 
-// fetch the JSON version data (only once), then use it to populate the version
-// switcher and maybe show the version warning bar
-var versionSwitcherBtns = document.querySelectorAll(
-  ".version-switcher__button",
-);
-const hasSwitcherMenu = versionSwitcherBtns.length > 0;
-const hasVersionsJSON = DOCUMENTATION_OPTIONS.hasOwnProperty(
-  "theme_switcher_json_url",
-);
-const wantsWarningBanner = DOCUMENTATION_OPTIONS.show_version_warning_banner;
-
-if (hasVersionsJSON && (hasSwitcherMenu || wantsWarningBanner)) {
-  const data = await fetchVersionSwitcherJSON(
```

Comment:
When I discovered this line I was a bit surprised. An `await` in the middle of a module file causes all the rest of the code below it to be deferred until the promise is settled. So this represents a pretty serious slowdown in executing this file, since pretty much all of the code in this file is actually run at the end of the file.

This was introduced in https://github.com/pydata/pydata-sphinx-theme/pull/1344. Previously, the fetch was also executed at the module level, but without the `await` syntax, so it didn't hold back the rest of the file from executing. 

So I decided to put all of this code into a new asynchronous function, `fetchAndUseVersions` which is called when the document is ready.